src/App.tsx

```diff
@@ -34,170 +187,130 @@ export const App: React.FC = () => {
         </header>
 
         <section className="todoapp__main" data-cy="TodoList">
-          {/* This is a completed todo */}
-          <div data-cy="Todo" className="todo completed">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-                checked
-              />
-            </label>
-
-            <span data-cy="TodoTitle" className="todo__title">
-              Completed Todo
-            </span>
+          {!loadingTodo &&
+            filteredTodos.map(todo => (
+              <div
+                key={todo.id}
+                data-cy="Todo"
+                className={`todo ${todo.completed ? 'completed' : ''}`}
+              >
+                {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+                <label
+                  className="todo__status-label"
+                  onClick={() => toggleCompleted(todo.id, todo.completed)}
+                >
+                  <input
+                    data-cy="TodoStatus"
+                    type="checkbox"
+                    className="todo__status"
+                    checked={todo.completed}
+                    readOnly
+                  />
+                </label>
 
-            {/* Remove button appears only on hover */}
-            <button type="button" className="todo__remove" data-cy="TodoDelete">
-              ×
-            </button>
+                {updatingId === todo.id ? (
+                  <form
+                    onSubmit={e => {
+                      e.preventDefault();
+                      handleSave(todo.id);
+                    }}
+                  >
+                    <input
+                      type="text"
+                      value={updatingText}
+                      onChange={e => setUpdatingText(e.target.value)}
+                      onBlur={() => handleSave(todo.id)}
+                      autoFocus
+                      className="todo__title-field"
+                      data-cy="TodoTitleField"
+                    />
+                  </form>
+                ) : (
```

Comment:
separate jsx and logic on different components (Header, Footer, TodoList, TodoItem...)